src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -346,7 +348,7 @@ def process_removal_queue(self) -> None:
         # self.mgr.to_remove_osds could change while this is processing (osds get added from the CLI)
         # The new set is: 'an intersection of all osds that are still not empty/removed (new_queue) and
         # osds that were added while this method was executed'
-        self.mgr.to_remove_osds.intersection_update(new_queue)
+        self.mgr.to_remove_osds.queue.intersection_update(new_queue)
```

Comment:
Shouldn't we encapsulate and lock this one too? Direct access to the queue container shouldn't be allowed.

Reply:
This is a bit of a weird one.. And that's probably due to the preceding logic. If we want to do this the _correct_(queue) way, we should compute the intersection and then re-add/delete it from the queue respectively. `put` and `rm` are however overloaded with custom code, which makes it weird. The whole queue processing section should become a revamp sooner or later..